homeassistant/components/climate/modbus.py

```diff
@@ -0,0 +1,167 @@
+import homeassistant.helpers.config_validation as cv
+
+DEPENDENCIES = ['modbus']
+
+# Parameters not defined by homeassistant.const
+CONF_TARGET_TEMP = 'target_temp_register'
+CONF_CURRENT_TEMP = 'current_temp_register'
+CONF_DATA_TYPE = 'data_type'
+CONF_COUNT = 'data_count'
+CONF_PRECISION = 'precision'
+
+DATA_TYPE_INT = 'int'
+DATA_TYPE_UINT = 'uint'
+DATA_TYPE_FLOAT = 'float'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_SLAVE): cv.positive_int,
+    vol.Required(CONF_TARGET_TEMP): cv.positive_int,
+    vol.Required(CONF_CURRENT_TEMP): cv.positive_int,
+    vol.Optional(CONF_DATA_TYPE, default=DATA_TYPE_FLOAT):
+        vol.In([DATA_TYPE_INT, DATA_TYPE_UINT, DATA_TYPE_FLOAT]),
+    vol.Optional(CONF_COUNT, default=2): cv.positive_int,
+    vol.Optional(CONF_PRECISION, default=1): cv.positive_int
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Modbus Thermostat Platform."""
+    name = config.get(CONF_NAME)
+    modbus_slave = config.get(CONF_SLAVE)
+    target_temp_register = config.get(CONF_TARGET_TEMP)
+    current_temp_register = config.get(CONF_CURRENT_TEMP)
+    data_type = config.get(CONF_DATA_TYPE)
+    count = config.get(CONF_COUNT)
+    precision = config.get(CONF_PRECISION)
+
+    add_devices([ModbusThermostat(hass, name, modbus_slave,
+                                  target_temp_register, current_temp_register,
+                                  data_type, count, precision)], True)
+
+
+class ModbusThermostat(ClimateDevice):
+    """Representation of a Modbus Thermostat."""
+
+    def __init__(self, hass, name, modbus_slave, target_temp_register,
+                 current_temp_register, data_type, count, precision):
+        """Initialize the unit."""
+        self.hass = hass
+        self._name = name
+        self._slave = modbus_slave
+        self._target_temperature_register = target_temp_register
+        self._current_temperature_register = current_temp_register
+        self._target_temperature = None
+        self._current_temperature = None
+        self._data_type = data_type
+        self._count = int(count)
+        self._precision = precision
+        self._structure = '>f'
+        self._error = False
+
+        data_types = {DATA_TYPE_INT: {1: 'h', 2: 'i', 4: 'q'},
+                      DATA_TYPE_UINT: {1: 'H', 2: 'I', 4: 'Q'},
+                      DATA_TYPE_FLOAT: {1: 'e', 2: 'f', 4: 'd'}}
+
+        self._structure = '>{}'.format(data_types[self._data_type]
+                                       [self._count])
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    def update(self):
+        """Update Target & Current Temperature."""
+        try:
+            result = modbus.HUB.read_holding_registers(
```

Comment:
Make a helper method that reads a register and parses the result. The helper should accept a register and return the parsed value, at minimum.

Use the helper to get target and current temperature.